Library/Homebrew/cask/installer.rb

```diff
@@ -119,6 +119,13 @@ def check_conflicts
       rescue CaskUnavailableError
         next # Ignore conflicting Casks that do not exist.
       end
+
+      @cask.conflicts_with[:formula].each do |conflicting_formula|
```

Comment:
I wonder if we already have logic for this somewhere else in the codebase. Also, is `#any_version_installed?` the right check or do we also have to check for `#opt_linked?`.